src/libraries/System.Drawing.Common/src/System/Drawing/DrawingComWrappers.cs

```diff
@@ -0,0 +1,279 @@
+            wrapperEntry->IID = Interop.Ole32.IStreamComWrapper.Guid;
+            wrapperEntry->Vtable = iStreamVtblRaw;
+            return wrapperEntry;
+        }
+
+        protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count)
+        {
+            Debug.Assert(obj is Interop.Ole32.IStreamComWrapper);
+            Debug.Assert(s_wrapperEntry != null);
+
+            // Always return the same table mappings.
+            count = 1;
+            return s_wrapperEntry;
+        }
+
+        protected override object CreateObject(IntPtr externalComObject, CreateObjectFlags flags)
+        {
+            Debug.Assert(flags == CreateObjectFlags.None);
+
+            Guid pictureGuid = IPicture.Guid;
+            int hr = Marshal.QueryInterface(externalComObject, ref pictureGuid, out IntPtr comObject);
+            if (hr == 0)
+            {
+                return new PictureWrapper(comObject);
+            }
+
+            throw new NotImplementedException();
+        }
+
+        protected override void ReleaseObjects(IEnumerable objects)
+        {
+            throw new NotImplementedException();
+        }
+
+        internal struct IUnknownVtbl
+        {
+            public IntPtr QueryInterface;
+            public IntPtr AddRef;
+            public IntPtr Release;
+        }
+
+        internal struct IStreamVtbl
+        {
+            public IUnknownVtbl IUnknownImpl;
+            public IntPtr Read;
+            public IntPtr Write;
+            public IntPtr Seek;
+            public IntPtr SetSize;
+            public IntPtr CopyTo;
+            public IntPtr Commit;
+            public IntPtr Revert;
+            public IntPtr LockRegion;
+            public IntPtr UnlockRegion;
+            public IntPtr Stat;
+            public IntPtr Clone;
+
+            private delegate void _Read(IntPtr thisPtr, byte* pv, uint cb, uint* pcbRead);
+            private delegate void _Write(IntPtr thisPtr, byte* pv, uint cb, uint* pcbWritten);
+            private delegate void _Seek(IntPtr thisPtr, long dlibMove, SeekOrigin dwOrigin, ulong* plibNewPosition);
+            private delegate void _SetSize(IntPtr thisPtr, ulong libNewSize);
+            private delegate void _CopyTo(IntPtr thisPtr, IntPtr pstm, ulong cb, ulong* pcbRead, ulong* pcbWritten);
+            private delegate void _Commit(IntPtr thisPtr, uint grfCommitFlags);
+            private delegate void _Revert(IntPtr thisPtr);
+            private delegate Interop.HRESULT _LockRegion(IntPtr thisPtr, ulong libOffset, ulong cb, uint dwLockType);
+            private delegate Interop.HRESULT _UnlockRegion(IntPtr thisPtr, ulong libOffset, ulong cb, uint dwLockType);
+            private delegate void _Stat(IntPtr thisPtr, out Interop.Ole32.STATSTG pstatstg, Interop.Ole32.STATFLAG grfStatFlag);
+            private delegate IntPtr _Clone(IntPtr thisPtr);
```

Comment:
We should be able to use `[UnmanagedCallersOnly]` and function pointers instead of delegates to reduce allocations and improve performance.